tests/test_r_regressions.py

```
@@ -0,0 +1,179 @@
+
+        # Create a file with a tricky name (quotes in path)
+        # Note: On most systems this is valid, but we test the escaping logic
+        r_file = tmp_path / "test_file.R"
+        r_file.write_text("x <- 1\n")
+
+        # Should not raise an exception
+        try:
+            result = get_diagnostics(str(r_file), "r", include_lint=True)
+            # Result may be empty if lintr not installed, but should not crash
+            assert isinstance(result, tuple)
+        except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)